server/controllers/manage/dateChangesController.ts

```diff
@@ -34,14 +35,27 @@ export default class DateChangeController {
     return async (req: Request, res: Response) => {
       const { premisesId, bookingId } = req.params
 
-      const payload = {} as NewDateChange
+      try {
+        const payload = {} as NewDateChange
+        const emptyDates = [] as Array<keyof NewDateChange>
         const datesToChange = [req.body.datesToChange || []].flat()
 
-        payload[itemKey] = DateFormats.dateAndTimeInputsToIsoString(req.body, itemKey)[itemKey]
-      })
         datesToChange.forEach((itemKey: keyof NewDateChange) => {
+          const date = DateFormats.dateAndTimeInputsToIsoString(req.body, itemKey)[itemKey]
```

Comment:
If this conversion fails will the catch handle the error properly?

Reply:
It should yeah, because any failure returns `undefined`